### DISCUSSION TITLE
How to strip empty paragraphs (at extension level)?

### DISCUSSION THREAD

**Discussion by ghost** [ueberdosis/tiptap#3904] [General Questions]
Is it possible to extend the Paragraph extension and determine if the paragraph is empty or not? In my app I want to strip empty paragraphs and I was wondering what’s the best way to do this …

When extending the `renderHTML` function/method I don’t have access to the content of a node (right?), because I only get this `0` thing …

**Comment by bdbch** [comment]
@JonathanDoelan you could create a Prosemirror plugin or add editor events for this. In the Prosemirror plugin for example you could hook into the update function and check for empty paragraphs to remove them.

Could be resource intensive on large documents though. Otherwise I don't have a specific solution in mind right now.

**Reply by ghost** [reply]
I don’t want to do it while updating the text, this could lead to bad user experience, I guess. But is there a way to do it while outputting (`generateHTML`) the content? With a Prosemirror plugin, do I have full access to the content?

**Reply by bdbch** [reply]
You have access to the full editor object in a plugin. That means you could access the HTML or JSON output by Prosemirror and modify / remap it.

Maybe you could give me an example on how you would like to do what you're trying to achieve so I can better understand what you want it to happen?

**Reply by ghost** [reply]
I am using a (headless) CMS where I have integrated the tiptap editor. Then I have a (Nuxt) frontend where I want to render the editor contents JSON to HTML. I don’t want to display empty paragraphs, because they break with the design on the frontend. So I need to strip these empty paragraphs. I don’t want to strip them in the CMS or while typing (or blurring), so I thought it would be the best, if I could strip these while generating the HTML output. So I thought I could extend the Paragraph extension and implement a few lines of code that will strip all these empty paragraphs …

An other option would be to strip the empty tags after the HTML is generated via regex — but I’m not sure which way is better.
